### PR TITLE
Make use of discord.VoiceProtocol instead of socket_response

### LINKED ISSUE #97: Use d.py's public interfaces (i.e. VoiceProtocol) instead of private ones
We should consider moving from the usage of `bot._connection._get_websocket` *and* `on_socket_response` to `discord.VoiceProtocol` and other supported methods that are now available in d.py.

### LINKED ISSUE #98: Does not work with discord.py v2.0
Discord.py v2 removes the `on_socket_response` event so the library won't work. Instead we will have the `on_raw_socket_receive` event.

### PR DESCRIPTION
**Pull request by PredaaA**
This PR makes use of `discord.VoiceProtocol` instead of listening to `socket_response` event. Which will be useful for next discord.py major release where that event no longer exists unless you pass a parameter to Client.
But it also makes things more "cleaner", as it "bind" the Red Lavalink `Player` object to `discord.Guild` object (`guild.voice_client`).
Though it comes with several breaking changes, that might break on current implementations of Red-Lavalink (in the case of Red's audio it doesn't seems to be a lot).

## Breaking changes:
- `player_manager.PlayerManager` no longer exists. The needed methods are now in `node.Node`.
- Therefore `Player.manager` no longer exists too, `node.Node` can be accessed instead. `Node.player_manager` too.
- `voice_ws_func` parameter is no longer needed to be passed in `node.Node`, and therefore the `get_voice_ws` method has been removed too.
- `player_manager.user_id` & `player_manager.channel_finder_func` are removed.
- `RESTClient` now takes `Player` instead of `Node` as parameter.
- The `rest_api.reset_session` & `rest_api.close` methods are removed.

Fixes #98 & closes #97.

### CODE REVIEW

**Comment by Jackenmen** [comment]
> It still make use of `bot._connection._get_websocket` though, maybe if there is an official API for that it would be better to use it

It seems we only use it here:
https://github.com/PredaaA/Red-Lavalink/blob/cff92370f2909f0796e662299c9e37cf46d71287/lavalink/player_manager.py#L246-L248

Which can probably be replaced with [`bot.is_closed()`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Client.is_closed).

**Comment by PredaaA** [comment]
> > It still make use of `bot._connection._get_websocket` though, maybe if there is an official API for that it would be better to use it
> 
> It seems we only use it here:
> [PredaaA/Red-Lavalink@`cff9237`/lavalink/player_manager.py#L246-L248](https://github.com/PredaaA/Red-Lavalink/blob/cff92370f2909f0796e662299c9e37cf46d71287/lavalink/player_manager.py#L246-L248)
> 
> Which can probably be replaced with [`bot.is_closed()`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Client.is_closed).

I ended up using [`Shard.is_closed()`](https://discordpy.readthedocs.io/en/stable/api.html#discord.ShardInfo.is_closed) instead, which seems to be doing the intended behavior (even on Red's cog side).
So the `get_voice_ws` method is removed too.

**Comment by FightMan01** [comment]
I'm testing this PR for 2 week now with my bot (9300+ guilds) and everything works fine. Thanks.

**Comment by PredaaA** [comment]
Sending that here so it won't be lost. There are mem leaks problems, and players state not being updated when stopped/disconnected.

File: lavalink/node.py [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
 from .rest_api import Track
+from .utils import VoiceChannel
 
-__all__ = ["Stats", "Node", "NodeStats", "get_node", "get_nodes_stats", "join_voice"]
+__all__ = ["Stats", "Node", "NodeStats", "get_node", "get_nodes_stats"]
 
 _nodes: List[Node] = []
 
-PlayerState = namedtuple("PlayerState", "position time")
+PositionTime = namedtuple("PositionTime", "position time")
```
Comment (aikaterna) [inline lavalink/node.py]: PositionTime = namedtuple("PositionTime", "position time", "connected")

File: lavalink/node.py [hunk 2]
```diff
@@ -367,7 +374,7 @@ async def _handle_op(self, op: LavalinkIncomingOp, data):
                 self.event_handler(op, event, data)
         elif op == LavalinkIncomingOp.PLAYER_UPDATE:
             state = data.get("state", {})
-            state = PlayerState(position=state.get("position", 0), time=state.get("time", 0))
+            state = PositionTime(position=state.get("position", 0), time=state.get("time", 0))
```
Comment (aikaterna) [inline lavalink/node.py]: state = PositionTime(
                position=state.get("position", 0),
                time=state.get("time", 0),
                connected=state.get("connected", False),
            )

File: lavalink/__init__.py [hunk 3]
```diff
@@ -43,4 +40,4 @@
     "all_connected_players",
     "active_players",
 ]
-__version__ = "0.8.1"
+__version__ = "0.9.0"
```
Comment (aikaterna) [inline lavalink/__init__.py]: Use 0.10.0 or whatever we end up with by the time this is merged

Also this needs to be moved above the imports like in https://github.com/Cog-Creators/Red-Lavalink/pull/103

**Comment by PredaaA** [comment]
Pushed changes from latest review that @aikaterna done and was pushed on another remote.

**Comment by PredaaA** [comment]
PR ready for review. (finally)

File: lavalink/__init__.py [hunk 4]
```diff
@@ -44,3 +41,4 @@
     "all_connected_players",
     "active_players",
 ]
+__version__ = "0.9.0"
```
Comment (Lifeismana) [inline lavalink/__init__.py]: Missed when phenom4n4n's branch was merged
(it's already there at line 4)
```suggestion
```

**Comment by aikaterna** [comment]
It should be 0.10.0

(nvm, I see what you're saying now Life)

**Comment by Lifeismana** [comment]
it's already there at line 4

**Comment by Jackenmen** [comment]
Superseded by #122

**Pull request closed without merging**